### PR TITLE
DFA-551 Smoke tests

### PR DESCRIPTION
**Pull request by JohnWattsGDS**
Add monitoring.

Entire canary stack can be deployed via SAM.  We only have one simple lambda so it's written inline in the template.  If we don't do that, we need to create a bucket, build and upload canaries and stuff.

Create `smoke-tests.yml` for a GitHub action that will deploy smoke tests if anything in `monitoring/` changes.
Create `github-actions-oidc-provider-template.yaml` so an OIDC provider can be created for a given AWS account.
Create `sam-deployment-artifacts-bucket-template.yaml` so a bucket can be created for SAM to deploy to.

### CODE REVIEW

**Review by CharlesIC — COMMENTED** [review]
Few small fixes

File: .github/workflows/smoke-tests.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: Smoke tests
+
+on:
+  push:
+    branches: [ DFA-551_smoke_tests ]
```
[the remark this answers is not in the document]
Comment (JohnWattsGDS) [reply .github/workflows/smoke-tests.yml:5]: 👍🏻

File: .github/workflows/smoke-tests.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+jobs:
+  deploy-canary-stack:
+    name: Deploy Canaries
+    environment: integration
+    runs-on: ubuntu-latest
+    env:
+      AWS_REGION: eu-west-2
+      STACK_NAME: di-onboarding-monitoring
+      ROLLBACK_ACTION: "--disable-rollback"
+      ACCOUNT: int
```
[the remark this answers is not in the document]
Comment (JohnWattsGDS) [reply .github/workflows/smoke-tests.yml:17]: As discussed, right here as a name-spacing prefix.

File: .github/workflows/smoke-tests.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - name: Checkout repository
```
[the remark this answers is not in the document]
Comment (JohnWattsGDS) [reply .github/workflows/smoke-tests.yml:24]: 👍🏻

File: .github/workflows/smoke-tests.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        uses: actions/checkout@v2
+
+      - name: Setup SAM
```
[the remark this answers is not in the document]
Comment (JohnWattsGDS) [reply .github/workflows/smoke-tests.yml:27]: :+1

File: .github/workflows/smoke-tests.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        uses: aws-actions/setup-sam@v1
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+          role-session-name: ${{ secrets.AWS_ROLE_SESSION }}
+          aws-region: ${{ env.AWS_REGION }}
```
[the remark this answers is not in the document]
Comment (JohnWattsGDS) [reply .github/workflows/smoke-tests.yml:35]: It is not 😄

File: .github/workflows/smoke-tests.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (JohnWattsGDS) [reply .github/workflows/smoke-tests.yml:27]: 👍🏻

File: ci/README.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# GithubActions OIDC Provider
+
+In order to use the GitHub OIDC provider to get credentials, you'll need an `AWS::IAM::OIDCProvider` set up.  The template `github-actions-role-template.yaml` contains the code provided by AWS to do so.  You only need one OIDC provider per account.
+
+You can run the command:
+
+```bash
+gds aws di-onboarding-<account> -- sam deploy --stack-name github-actions-oidc-provider --template-file ci/github-actions-oidc-provider-template.yaml --capabilities CAPABILITY_IAM --parameter-overrides "GitHubOrg=alphagov RepositoryName=di-onboarding-product-page"
```
[the remark this answers is not in the document]
Comment (JohnWattsGDS) [reply ci/README.md:8]: 👍🏻

File: ci/README.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+```yaml
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+          role-session-name: ${{ secrets.AWS_ROLE_SESSION }}
+          aws-region: ${{ env.AWS_REGION }}
+```
+
+See `.github/workflows/smoke-tests.tml` for an example.
```
[the remark this answers is not in the document]
Comment (JohnWattsGDS) [reply ci/README.md:24]: 👍🏻 it does

**Review by CharlesIC — APPROVED** [review]

**Pull request merged**